src/js/init-favorites-buttons.js

```diff
@@ -3,36 +3,69 @@ import { constants } from './constants';
 let addFavoritesButton;
 let removeFavoritesButton;
 
-export function initFavoritesButtons() {
+export function initFavoritesButtons(exerciseId) {
   addFavoritesButton = document.getElementById('addFavoritesButton');
   removeFavoritesButton = document.getElementById('removeFavoritesButton');
   addFavoritesButton.addEventListener('click', addToFavorites);
   removeFavoritesButton.addEventListener('click', removeFromFavorites);
-  const favorites = JSON.parse(localStorage.getItem(constants.FAV_KEY)) ?? [];
-  const exerciseId = addFavoritesButton.dataset.id;
-  if (exerciseId && favorites.includes(exerciseId)) {
+
+  if (getExerciseDataById(exerciseId)) {
     showRemoveFavoritesButton();
   } else {
     showAddFavoritesButton();
   }
 }
 
+function parseFavoritesData() {
+  const modal = document.getElementById('exerciseModal');
+  const _id = modal.querySelector('#addFavoritesButton').getAttribute('data-id');
+  const name = modal.querySelector('.exercise-header h2').textContent;
+  const rating = parseFloat(modal.querySelector('.rating-value').textContent);
+  const target = modal.querySelector('.exercise-info .details-column:nth-child(1) .details-value').textContent;
+  const bodyPart = modal.querySelector('.exercise-info .details-column:nth-child(2) .details-value').textContent;
+  const burnedCaloriesText = modal.querySelector('.exercise-info .details-column:nth-child(5) .details-value').textContent;
+  const burnedCalories = parseInt(burnedCaloriesText.split('/')[0]);
+  const timeText = burnedCaloriesText.split('/')[1];
+  const time = parseInt(timeText.split(' ')[0]);
+
+  return {
+      _id,
+      name,
+      burnedCalories,
+      rating,
+      target,
+      time,
+      bodyPart
+  };
+}
+
+function storeExerciseData(exerciseData) {
+  let favorites = JSON.parse(localStorage.getItem(constants.FAV_KEY)) || {};
+  favorites[exerciseData._id] = exerciseData;
+  localStorage.setItem('favorites', JSON.stringify(favorites));
+}
+
+function getExerciseDataById(id) {
+  const favorites = JSON.parse(localStorage.getItem(constants.FAV_KEY)) || {};
+  return favorites[id] || null;
+}
+
 function addToFavorites() {
-  const favorites = JSON.parse(localStorage.getItem(constants.FAV_KEY)) ?? [];
-  const exerciseId = addFavoritesButton.dataset.id;
-  if (exerciseId && !favorites.includes(exerciseId)) {
-    favorites.push(addFavoritesButton.dataset.id);
-    localStorage.setItem(constants.FAV_KEY, JSON.stringify(favorites));
+
+  const data = parseFavoritesData();
+  const favoritesRecord = getExerciseDataById(data._id);
+  if (favoritesRecord == null) {
```

Comment:
```suggestion
  if (!favoritesRecord) {
```
чи має бути перевірка саме на `null` і `false` наприклад не має проходити?